errors.go

```diff
@@ -280,3 +280,34 @@ func Cause(err error) error {
 	}
 	return err
 }
+
+// Stack returns the first stack trace of the errors, if possible.
+func Stack(err error) StackTrace {
+	type causer interface {
+		Cause() error
+	}
+
+	type stackTracer interface {
+		StackTrace() StackTrace
+	}
+
+	var stackErr error
```

Comment:
We could avoid having to re-type-assert the stackTracer if we recognize that `stackErr` is always going to be a `stackTracer` and enforce that through typing:
```
	var top stackTracker
	…
		if stack, ok := err.(stackTracer); ok {
			top = stack
		}
	…
	if top == nil {
		return nil
	}

	return top.StackTrace()
```